src/components/Toolbar.tsx

```diff
@@ -94,21 +121,43 @@ const Toolbar: FC<ToolbarProps> = ({ customize, onSelect, selected }) => {
     }
   }, [isDraggingAny, setPressingToolbarId])
 
+  /** Handles the state of icon transitions, forcing completion of icon animation. */
   useEffect(() => {
     /**
-     * Update isPressed when the drag is released outside of the toolbar.
-     * */
-    const handleMouseUp = event => {
-      setPressingToolbarId(null)
+     *Function to handle the transition end event.
+     **/
+    const handleTransitionEnd = () => {
+      if (!isMousePressed) {
+        setPressingToolbarId(null)
+      }
+      setIsTransitioning(false)
     }
 
+    // Select all elements with the class 'toolbar-icon'
+    const icons = document.querySelectorAll('.toolbar-icon')
```

Comment:
It's usually best to avoid directly accessing the DOM from React components unless absolutely necessary, as changes to the DOM are not captured by the type system.

If we go with this, however, the approach can be slightly improved by using a separate data attribute rather than reusing an existing CSS class.